File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Dodge.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Dodge implements Listener {
    @EventHandler
    public void onDamage(final EntityDamageEvent entityDamageEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "dodge"));
        if (entityDamageEvent.getEntity() instanceof Player) {
            final Player player = (Player) entityDamageEvent.getEntity();
            final int n = (int) (Math.random() * 100.0);
            final ItemStack[] armorContents = player.getInventory().getArmorContents();
            try {
                ItemStack[] array;
                for (int length = (array = armorContents).length, i = 0; i < length; ++i) {
                    final ItemStack itemStack = array[i];
                    if (itemStack != null && itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && n < SettingsManager.enchant.getInt("dodge.level_I.chance")) {
                        entityDamageEvent.setDamage(0.0);
                    }
                }
                ItemStack[] array2;
                for (int length2 = (array2 = armorContents).length, j = 0; j < length2; ++j) {
                    final ItemStack itemStack2 = array2[j];
                    if (itemStack2 != null && itemStack2.hasItemMeta() && itemStack2.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II") && n < SettingsManager.enchant.getInt("dodge.level_II.chance")) {
                        entityDamageEvent.setDamage(0.0);
                    }
                }
                ItemStack[] array3;
                for (int length3 = (array3 = armorContents).length, k = 0; k < length3; ++k) {
                    final ItemStack itemStack3 = array3[k];
                    if (itemStack3 != null && itemStack3.hasItemMeta() && itemStack3.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III") && n < SettingsManager.enchant.getInt("dodge.level_III.chance")) {
                        entityDamageEvent.setDamage(0.0);
                    }
                }
                ItemStack[] array4;
                for (int length4 = (array4 = armorContents).length, l = 0; l < length4; ++l) {
                    final ItemStack itemStack4 = array4[l];
                    if (itemStack4 != null && itemStack4.hasItemMeta() && itemStack4.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV") && n < SettingsManager.enchant.getInt("dodge.level_IV.chance")) {
                        entityDamageEvent.setDamage(0.0);
                    }
                }
                ItemStack[] array5;
                for (int length5 = (array5 = armorContents).length, n2 = 0; n2 < length5; ++n2) {
                    final ItemStack itemStack5 = array5[n2];
                    if (itemStack5 != null && itemStack5.hasItemMeta() && itemStack5.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " V") && n < SettingsManager.enchant.getInt("dodge.level_V.chance")) {
                        entityDamageEvent.setDamage(0.0);
                    }
                }
            } catch (Exception ex) {
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantments/Dodge.kt
package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Dodge : Listener {
    private val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "dodge"))
    @EventHandler
    fun onDamage(entityDamageEvent: EntityDamageEvent) {
        if (entityDamageEvent.entity is Player) {
            val player = entityDamageEvent.entity as Player
            val n = (Math.random() * 100.0).toInt()
            val armorContents = player.inventory.armorContents + IM.getAccessorySlots(player)
            try {
                for (i in armorContents.indices) {
                    val itemStack = armorContents[i]
                    if (itemStack != null && itemStack.hasItemMeta()) {
                        val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
                        if (level > 0 && (Math.random() * 100.0).toInt() < SettingsManager.enchant.getInt("dodge.$level.chance")) {
                            entityDamageEvent.damage = 0.0
                        }
                    }
                }
            } catch (ex: Exception) {
            }

        }
    }
}
